src/prisma/schedule.ts

```diff
@@ -37,3 +39,93 @@ async function initiateSchedule(scheduleyear: number) {
   });
   return schedule;
 }
+
+async function updateCurrentSchedule(
+  scheduleId: any,
+  courses: CourseSectionInput[]
+) {
+  // Simply creates a new schedule instead of updating the current schedule
+
+  let currentSchedule;
+
+  try {
+    currentSchedule = await prisma.schedule.findFirst({
+      where: {
+        id: parseInt(scheduleId),
+      },
+    });
+  } catch (error) {
+    return {
+      success: false,
+      message: "Couldn't find schedule",
+      errors: error,
+    };
+  }
+
+  let newSchedule;
+
+  try {
+    newSchedule = await prisma.schedule.create({
+      data: {
+        year: currentSchedule?.year ?? 2022,
+      },
+    });
+  } catch (error) {
+    return {
+      success: false,
+      message: 'Error creating new update schedule',
+      errors: error,
+    };
+  }
+
+  try {
+    for (const course of courses) {
+      // Rename day to days for prisma
+      const newMeetingTimes = course.meetingTimes.map((meeting) => ({
+        startTime: meeting.startTime,
+        endTime: meeting.endTime,
+        days: meeting.day,
+      }));
+
+      await prisma.courseSection.create({
+        data: {
+          hoursPerWeek: course.hoursPerWeek ?? 3,
+          sectionNumber: course.sectionNumber ?? '',
+          capacity: course.capacity ?? 100,
+          startDate: course.startDate,
+          endDate: course.endDate,
+          user: {
+            connect: {
+              username: course.professors[0] ?? '',
+            },
+          },
+          meetingTime: {
+            createMany: {
+              data: newMeetingTimes,
+            },
+          },
+          course: {
+            connect: {
+              subject_courseCode_term: {
+                subject: course.id.subject,
+                courseCode: course.id.code,
+                term: course.id.term,
+              },
+            },
+          },
+          schedule: {
+            connect: {
+              id: newSchedule.id,
+            },
+          },
+        },
+      });
+    }
+  } catch (error) {}
```

Comment:
Could this loop be replaced with a `createMany`?

Reply:
![image](https://user-images.githubusercontent.com/47232534/179856912-3d7c769a-d139-43de-ba92-1d3a64f06db6.png)

I was facing this issue when using createMany. Do you have any suggestions you know of that would work better?

Reply:
I can't figure it out. I looked into a bit but I think what you have here is the best we can do 👍